File: backend/src/main/java/com/tutr/api/lessons/Lesson.java
package com.tutr.api.lessons;

import com.tutr.api.common.BaseEntity;
import com.tutr.api.students.Student;
import com.tutr.api.users.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "lessons")
public class Lesson extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tutor_id", nullable = false)
    private User tutor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_series_id")
    private LessonSeries lessonSeries;

    private String title;

    @Column(nullable = false)
    private Instant lessonDate;

    @Column(nullable = false)
    private Integer durationMinutes;

    @Column(nullable = false)
    private BigDecimal hourlyRate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LessonStatus status = LessonStatus.SCHEDULED;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus paymentStatus = PaymentStatus.UNPAID;

    private String lessonNotes;
    private String homework;
    private String miroBoardUrl;
    private String inviteEmail;

    private String googleEventId;
    private String googleCalendarId;
    private String googleMeetLink;
    private boolean googleSyncEnabled;

    @Enumerated(EnumType.STRING)
    private GoogleSyncStatus googleSyncStatus = GoogleSyncStatus.NOT_REQUESTED;

    private String googleSyncError;
}

File: frontend/src/types/lesson.ts
export type LessonStatus = 'SCHEDULED' | 'COMPLETED' | 'CANCELLED' | 'NO_SHOW';
export type PaymentStatus = 'UNPAID' | 'PAID' | 'PARTIAL';
export type GoogleSyncStatus = 'NOT_REQUESTED' | 'NOT_CONNECTED' | 'SYNCED' | 'FAILED';
export type RecurringFrequency = 'WEEKLY';

export type Lesson = {
  id: string;
  lessonSeriesId?: string | null;
  studentId: string;
  studentName: string;
  title?: string;
  lessonDate: string;
  durationMinutes: number;
  hourlyRate: number;
  status: LessonStatus;
  paymentStatus: PaymentStatus;
  lessonNotes?: string;
  homework?: string;
  miroBoardUrl?: string;
  inviteEmail?: string;
  googleColorId?: string | null;
  googleExtraReminderMinutes?: number | null;
  googleSyncEnabled?: boolean;
  googleMeetLink?: string | null;
  googleSyncStatus?: GoogleSyncStatus;
  googleSyncError?: string | null;
  createdAt: string;
};

export type LessonPayload = Omit<Lesson, 'id' | 'studentName' | 'lessonSeriesId' | 'googleMeetLink' | 'googleSyncStatus' | 'googleSyncError' | 'createdAt'> & {
  syncToGoogle?: boolean;
};

export type LessonStatusPayload = {
  status?: LessonStatus;
  paymentStatus?: PaymentStatus;
};

export type RecurringLessonPayload = {
  studentId: string;
  title?: string;
  firstLessonDate: string;
  durationMinutes: number;
  hourlyRate: number;
  frequency: RecurringFrequency;
  intervalCount?: number;
  recurrenceUntil?: string;
  lessonNotes?: string;
  homework?: string;
  miroBoardUrl?: string;
  inviteEmail?: string;
  googleColorId?: string | null;
  googleExtraReminderMinutes?: number | null;
  syncToGoogle?: boolean;
};
